ClientApp/viking-vault/src/components/TopBar.tsx

```diff
@@ -1,13 +1,58 @@
 import React from 'react'
 import './styles.css'
 import logo from './images/logo.png'
+import { Link } from 'react-router-dom';
+import {variables} from "../ConstantVariables";
 
-class TopBar extends React.Component{
+const baseUrl = variables.baseUrl;
+
+interface IState {
+    isAdmin: boolean;
+}
+
+class TopBar extends React.Component<any, IState>{
```

Comment:
Please use types for state and props :)

Reply:
But the component has no props. We don't need props for this component